osgi.specs/docbook/core/003/framework.module.xml

```diff
@@ -2229,12 +2240,44 @@ package-names   ::= package-name                // See <xref
       clean API-implementation separation. OSGi services are carefully
       designed to be as standalone as possible. Many libraries intertwine API
       and implementation in the same package making it impossible to
-      substitute the API packages.</para>
+      substitute the API packages.
+      Developers of general-purpose Java libraries that include OSGi metadata, 
+      but were not originally designed for OSGi, should avoid importing exported packages.
+      </para>
 
-      <para>Importing an exported package must use a version range according
+      <para>Importing an exported package <emphasis role="bold">must</emphasis> use a version range according
       to its compatibility requirements, being either a consumer or a provider
       of that API. See <xref linkend="framework.module-semantic.versioning"/>
       for more information.</para>
+
+      <para>Also see the section about <xref linkend="framework.module-import.export.same.package"/> 
+      for more information about how the resolver handles this and what problems could arise with substitution.</para>
+
+      <note>
+      <para>
+		  Bundles should only import such an exported package if they are willing to get that package 
+		  from another (unknown) bundle and omit their own export from the framework. By definition, this means that a bundle with a subsitutable export may never load that package, rendering its class files and resources invisible to that bundle's class loader.
+      </para>
+      
+      <para>
+      Multiple bundles providing the same API packages 
+      can lead to conflicts in non-OSGi environments.
+      In OSGi environments these scenarios can often be supported, but resolution conflicts can still occur
+      when multiple versions of the same bundle are present (see <xref linkend="framework.module-import.export.same.package"/>).
```

Comment:
```suggestion
      Even though OSGi supports these scenarios, it might lead to resolution conflicts can when multiple versions of the same bundle are present (see <xref linkend="framework.module-import.export.same.package"/>) and consumers are having tight version range constraints or large dependency chains.
```